### PR TITLE
Validation support

### PR DESCRIPTION
**Pull request by shariq1501**
- Fixed Rule set validation
- Fixed change loading message and formate working actions

### CODE REVIEW

File: app/src/main/java/com/nerdapplabs/msoauth2/utility/Utility.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+public class Utility {
+
+    public static int getYearBeforeHundredYears() {
+        Calendar prevYear = Calendar.getInstance();
+        prevYear.add(Calendar.YEAR, -105);
+        return prevYear.get(Calendar.YEAR);
+    }
+
+    public static int getYearBeforeFiveYears() {
```
Comment (praveen-garg) [inline app/src/main/java/com/nerdapplabs/msoauth2/utility/Utility.java]: getDobEndYear

File: app/src/main/java/com/nerdapplabs/msoauth2/utility/Utility.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+import java.util.Calendar;
+
+/**
+ * Created by Mohd. Shariq on 21/02/17.
+ */
+
+public class Utility {
+
+    public static int getYearBeforeHundredYears() {
```
Comment (praveen-garg) [inline app/src/main/java/com/nerdapplabs/msoauth2/utility/Utility.java]: getDobStartYear

File: app/src/main/res/layout/activity_change_password.xml [hunk 3]
```diff
@@ -33,6 +33,7 @@
                     android:layout_width="match_parent"
                     android:layout_height="wrap_content"
                     android:hint="@string/old_password"
+                    android:maxLength="@integer/password_max_length"
```
Comment (praveen-garg) [inline app/src/main/res/layout/activity_change_password.xml]: max or min?

**Comment by shariq1501** [comment]
All issues fixed. Merging

**Pull request merged**